homeassistant/components/ssdp/__init__.py

```diff
@@ -7,14 +7,15 @@
 from enum import Enum
 from ipaddress import IPv4Address, IPv6Address
 import logging
-from typing import Any, Callable, Mapping
+from typing import Any, Callable, Final, Mapping, cast
 
 from async_upnp_client.aiohttp import AiohttpSessionRequester
 from async_upnp_client.const import DeviceOrServiceType, SsdpHeaders, SsdpSource
 from async_upnp_client.description_cache import DescriptionCache
 from async_upnp_client.ssdp import SSDP_PORT
 from async_upnp_client.ssdp_listener import SsdpDevice, SsdpListener
 from async_upnp_client.utils import CaseInsensitiveDict
+from typing_extensions import TypedDict
```

Comment:
Note: this is importing `TypedDict` from `typing_extensions` not `typing` to avoid `pylint(inherit-non-class)` issue.

Reply:
Given HA requires Python 3.8, is this still necessary?

Reply:
As I mentioned above, if I don't import it from typing_extensions `pylint` complains.
This is an extract from `typing_extensions`:
```python
if sys.version_info >= (3, 9, 2):
    # The standard library TypedDict in Python 3.8 does not store runtime information
    # about which (if any) keys are optional.  See https://bugs.python.org/issue38834
    # The standard library TypedDict in Python 3.9.0/1 does not honour the "total"
    # keyword with old-style TypedDict().  See https://bugs.python.org/issue42059
    TypedDict = typing.TypedDict
```

Reply:
That's unfortunate about the bug. I think you're right in that this is the best work-around.

Should we add a comment to the source stating why it's there, and when it can use the normal `TypedDict`? Perhaps a core reviewer can say whether such comments are good/bad in HA source?